iceoryx_posh/include/iceoryx_posh/internal/popo/modern_api/base_publisher.inl

```diff
@@ -22,8 +22,9 @@ namespace iox
 namespace popo
 {
 template <typename T, typename port_t>
-inline BasePublisher<T, port_t>::BasePublisher(const capro::ServiceDescription& service)
-    : m_port(iox::runtime::PoshRuntime::getInstance().getMiddlewarePublisher(service))
+inline BasePublisher<T, port_t>::BasePublisher(const capro::ServiceDescription& service,
+                                               const PublisherOptions& publisherOptions)
+    : m_port(iox::runtime::PoshRuntime::getInstance().getMiddlewarePublisher(service, publisherOptions.historyCapacity))
```

Comment:
According to lcov these lines are not covered. Could you add a test case here?

Reply:
I just had a look at it and I noticed that I need to mock the `PoshRuntime`. Alternatively, the RouDiEnvironment could be used, but it's quite heavy weight for a module tests and I think a `PoshRuntimeMock` could help with many other tests. It's quite some work though and if you don't mind, I would create a separate issue for this.

Reply:
adde #449